src/Core/Framework/Bundle.php

```diff
@@ -112,15 +126,25 @@ private function registerEvents(ContainerBuilder $container): void
         $definition->addMethodCall('addMultiple', [$this->getActionEvents()]);
     }
 
+    /**
+     * Looks for service definition files inside the `Resources/config`
+     * directory and loads the first one, it finds.
+     *
+     * @param ContainerBuilder $container
+     */
     private function registerContainerFile(ContainerBuilder $container): void
     {
         $fileSystem = new Filesystem();
-        $containerFilePath = 'Resources/config/services.xml';
-        if (!$fileSystem->exists($this->getPath() . '/' . $containerFilePath)) {
-            return;
-        }
 
-        $loader = new XmlFileLoader($container, new FileLocator($this->getPath()));
-        $loader->load($containerFilePath);
+        foreach (self::SUPPORTED_CONTAINER_FILES as $path => $loaderClass) {
```

Comment:
you could use the DelegatingLoader for that one:
```php
$loaderResolver = new LoaderResolver([
    new XmlFileLoader($container, new FileLocator($this->getPath()),
    new YamlFileLoader($container, new FileLocator($this->getPath()),
]);
$delegatingLoader = new DelegatingLoader($loaderResolver);

$delegatingLoader->load($containerFilePath);
```

Reply:
Could you explain the benefit? I would need to iterate either way, just to target the right filepath that the delegating loader needs to resolve a loader, right?

Reply:
i personally would probably go for something like this:
```
$loaderResolver = new LoaderResolver([
    new XmlFileLoader($container, new FileLocator($this->getPath()),
    new YamlFileLoader($container, new FileLocator($this->getPath()),
]);
$delegatingLoader = new DelegatingLoader($loaderResolver);

foreach (glob($this->getPath() . '/Resources/config/services.*' as $
    $delegatingLoader->load($containerFilePath);
}
```

would be easily extendable, the one problem i see in your implementation is the 
```
$loader = new $loaderClass($container, new FileLocator($this->getPath()));
```
line, because the constructor is in no interface or something, IMHO we should not depend on that these constructor arguments won't change.

The main problem is that the dynamic constructor call is not statically analyzable and fixable.

Reply:
I get your point and it sounds reasonable. I am only a bit concerned, if this is worth the extra allocations. But then again, the services are only loaded once.

I'm gonna rework this and change the PR.